src/components/VmActions/ConsoleButton.js

```diff
@@ -109,28 +121,31 @@ class ConsoleButton extends React.Component {
 
 ConsoleButton.propTypes = {
   vm: PropTypes.object.isRequired,
-  VmAction: PropTypes.object.isRequired,
-  usbFilter: PropTypes.string.isRequired, // eslint-disable-line react/no-unused-prop-types
-  consoleId: PropTypes.string, // eslint-disable-line react/no-unused-prop-types
-  className: PropTypes.string.isRequired,
-  tooltip: PropTypes.string,
+
+  isOnCard: PropTypes.bool,
+  actionDisabled: PropTypes.bool,
   shortTitle: PropTypes.string.isRequired,
+  tooltip: PropTypes.string,
   button: PropTypes.string.isRequired,
-  actionDisabled: PropTypes.bool,
-  isOnCard: PropTypes.bool,
-  userId: PropTypes.string, // eslint-disable-line react/no-unused-prop-types
-  onDownloadConsole: PropTypes.func.isRequired,
-  onConsoleSessionConfirmaClose: PropTypes.func.isRequired,
+  className: PropTypes.string.isRequired,
+
+  consoleId: PropTypes.string,
+  usbFilter: PropTypes.string,
+  userId: PropTypes.string,
+
   onCheckConsoleSessionInUse: PropTypes.func.isRequired,
+  onConsoleSessionConfirmClose: PropTypes.func.isRequired,
+  onDownloadConsole: PropTypes.func.isRequired,
 }
 
 export default connect(
   (state) => ({
-    VmAction: state.VmAction,
+    usbFilter: state.config.get('usbFilter'),
+    userId: state.config.getIn(['user', 'id']),
   }),
-  (dispatch, { vm, consoleId, usbFilter, userId }) => ({
-    onCheckConsoleSessionInUse: () => dispatch(checkConsoleInUse({ vmId: vm.get('id'), usbFilter, userId })),
-    onConsoleSessionConfirmaClose: () => dispatch(setConsoleInUse({ vmId: vm.get('id'), consoleInUse: false })),
-    onDownloadConsole: () => dispatch(downloadConsole({ vmId: vm.get('id'), consoleId, usbFilter })),
+  (dispatch) => ({
```

Comment:
Why did you remove `{ vm, consoleId }`, if in any case this is `ownProp`?

Reply:
I removed them to be consistent with all of the variables going to the dispatch actions.

Reply:
But it's not needed, because if you have some component, that represent Console, it has by default some props that define this component, in this case it is `vmId` and `consoleId` (not in every case). Thus you don't need to put them as argument to those function directly. The code will be more readable and it will be more consistent with the rest of code then now.

Reply:
I'll agree to disagree on this.  Either way the info is getting to the actions in the dispatch.

Perhaps you can write up a few of these web-ui coding best practices and put them in README or similar so other devs don't have to go through the same comment stream in the future.

Reply:
> Perhaps you can write up a few of these web-ui coding best practices and put them in README or similar so other devs don't have to go through the same comment stream in the future.

Actually, yes, I had thoughts about some documentation and we already have issue for this https://github.com/oVirt/ovirt-web-ui/issues/99. Soon I get to this.

Reply:
I'll need this too as I'm trying to catch my skills up